rust/agents/scraper/src/db/block.rs

```diff
@@ -96,11 +94,20 @@ impl ScraperDb {
             .collect::<Vec<_>>();
 
         debug_assert!(!models.is_empty());
-        let id_offset = models.len() as i64 - 1;
         debug!(blocks = models.len(), "Writing blocks to database");
         trace!(?models, "Writing blocks to database");
-        let first_id = Insert::many(models).exec(&self.0).await?.last_insert_id - id_offset;
-        debug_assert!(first_id > 0);
-        Ok(first_id)
+        match Insert::many(models)
+            .on_conflict(
+                OnConflict::column(block::Column::Hash)
+                    .do_nothing()
+                    .to_owned(),
+            )
```

Comment:
Is `on_conflict` needed here to avoid the panic of multiple insertion attempts for the same block?

Reply:
See other explanation, but basically, we can either insert one at a time or just ignore duplicates. All this is saying is "If you already know this hash, disregard this specific block" It will however still insert all the ones that do not conflict.

This is required if multiple scrapers are running against the same database for the same chain, but this is not actually a supported configuration because there is no reason to do that. The only way to go back to how it was is to implement #2631